src/services/superfluidMinionService.js

```diff
@@ -0,0 +1,142 @@
+import Web3 from 'web3';
+
+import SuperfluidMinionAbi from '../contracts/superfluidMinion.json';
+import { SUPERFLUID_MINION_STREAMS } from '../graphQL/superfluid-queries';
+import { TokenService } from './tokenService';
+import { chainByID, getGraphEndpoint } from '../utils/chain';
+import { graphFetchAll } from '../utils/theGraph';
+
+import SuperfluidSDK from '@superfluid-finance/js-sdk';
+
+let sf = null;
+
+const getSuperTokenBalances = async (chainID, minion, superTokens) => {
+  const tokenBalances = superTokens.map(async (tokenAddress) => {
+    const tokenService = TokenService({
+      tokenAddress: tokenAddress,
+      chainID,
+    });
+    return {
+      [tokenAddress]: {
+        tokenBalance: await tokenService('balanceOf')(minion),
+        symbol: await tokenService('symbol')(minion),
+        decimals: await tokenService('decimals')(minion),
+      },
+    };
+  });
+  return Object.assign({}, ...(await Promise.all(tokenBalances)));
+};
+
+export const SuperfluidMinionService = ({ web3, minion, chainID }) => {
+  const chainConfig = chainByID(chainID);
+  if (!web3) {
+    const rpcUrl = chainConfig.rpc_url;
+    web3 = new Web3(new Web3.providers.HttpProvider(rpcUrl));
+  }
+
+  const contract = new web3.eth.Contract(SuperfluidMinionAbi, minion);
+
+  return function getService(service) {
+    if (service === 'fetchStreams') {
+      return async ({ molochAddress }) => {
+        const streams = await graphFetchAll({
+          endpoint: getGraphEndpoint(chainID, 'subgraph_url'),
+          query: SUPERFLUID_MINION_STREAMS,
+          variables: {
+            minionId: `${molochAddress}-minion-${minion}`,
+          },
+          subfield: 'minionStreams',
+        });
+        if (!streams.length) {
+          return {
+            flows: [],
+            superTokens: null,
+          };
+        }
+        const superTokens = await getSuperTokenBalances(
+          chainID,
+          minion,
+          Array(
+            ...new Set(
+              streams.filter((s) => s.executed).map((s) => s.superTokenAddress),
+            ),
+          ),
+        );
+        if (!sf) {
+          // load SuperfluidSDK
+          const sfVersion = chainByID(chainID).superfluid_version;
+          sf = new SuperfluidSDK.Framework({
+            web3: web3,
+            version: sfVersion,
+          });
+          await sf.initialize();
+        }
+        const now = new Date();
+        const flows = await Promise.all(
+          streams.map(async (stream) => {
+            if (stream.executed) {
+              const decimals = superTokens[stream.superTokenAddress].decimals;
+              if (stream.active) {
+                // flow -> { timeStamp, flowRate, deposit, owedDeposit }
+                const flow = await sf.cfa.getFlow({
+                  superToken: stream.superTokenAddress,
+                  sender: minion,
+                  receiver: stream.to,
+                });
+                flow.netFlow =
```

Comment:
Avoid mutation

Reply:
this will be reviewed again in the next PR update